apps/src/dance/Dance.js

```diff
@@ -590,8 +592,22 @@ Dance.prototype.initInterpreter = function () {
     },
   };
 
+  const studentCode = this.studioApp_.getCode();
+
+  // Process studentCode to determine which characters are referenced and update
+  // this.charactersReferenced array with the results:
+  this.charactersReferenced = [];
+  const charactersRegExp = new RegExp(/^.*makeNewDanceSprite(?:Group)?\([^"]*"([^"]*)[^\r\n]*/, 'gm');
+  let match;
+  while ((match = charactersRegExp.exec(studentCode))) {
+    const characterName = match[1];
+    if (!this.charactersReferenced.includes(characterName)) {
+      this.charactersReferenced.push(characterName);
```

Comment:
Let's use `Set` instead of doing our own duplicate-check.

Reply:
Yes, I considered that. Happy to see us change to that.

Reply:
Fixed to use `Set`. Also, per comment above, eliminated `charactersReferenced` as an instance variable. The array is now returned from `initInterpreter()`.